src/bit-systems/media-loading.ts

```diff
@@ -179,36 +182,50 @@ function* loadByMediaType(
   //       content then using MediaImage/VideoLoaderData as like
   //       transporting data from the inflators. This may be like
   //       special and a bit less maintainable.
+  let mediaEid;
   switch (mediaType) {
     case MediaType.IMAGE:
-      return yield* loadImage(
+      mediaEid = yield* loadImage(
         world,
         accessibleUrl,
         contentType,
         MediaImageLoaderData.has(eid) ? MediaImageLoaderData.get(eid)! : {}
       );
+      break;
     case MediaType.VIDEO:
-      return yield* loadVideo(
+      mediaEid = yield* loadVideo(
         world,
         accessibleUrl,
         contentType,
         MediaVideoLoaderData.has(eid) ? MediaVideoLoaderData.get(eid)! : {}
       );
+      break;
     case MediaType.MODEL:
-      return yield* loadModel(world, accessibleUrl, contentType, true);
+      mediaEid = yield* loadModel(world, accessibleUrl, contentType, true);
+      break;
     case MediaType.PDF:
-      return yield* loadPDF(world, accessibleUrl);
+      mediaEid = yield* loadPDF(world, accessibleUrl);
+      break;
     case MediaType.AUDIO:
-      return yield* loadAudio(
+      mediaEid = yield* loadAudio(
         world,
         accessibleUrl,
         MediaVideoLoaderData.has(eid) ? MediaVideoLoaderData.get(eid)! : {}
       );
+      break;
     case MediaType.HTML:
-      return yield* loadHtml(world, canonicalUrl, thumbnail);
+      mediaEid = yield* loadHtml(world, canonicalUrl, thumbnail);
+      break;
     default:
       throw new UnsupportedMediaTypeError(eid, mediaType);
   }
+
+  if (hasComponent(world, MediaLink, eid)) {
+    inflateLink(world, mediaEid, { href: APP.getString(MediaLink.src[eid])!, type: LinkType.LINK });
+    inflateGrabbable(world, mediaEid, { cursor: true, hand: false });
+  }
```

Comment:
`MediaLink` is designed to be used only in media loading so it would be better to remove it when `MediaLoader` component is removed. Maybe in `mediaLoaderExitQuery` would be the best place.

Reply:
Absolutely https://github.com/mozilla/hubs/pull/6327

Thanks.